.buildkite/test-pipeline.yaml

```diff
@@ -469,11 +465,25 @@ steps:
   - pytest models/encoder_decoder/vision_language/test_broadcast.py -v -s -m 'distributed(num_gpus=2)'
   - pytest models/decoder_only/vision_language/test_models.py -v -s -m 'distributed(num_gpus=2)'
   - pytest -v -s spec_decode/e2e/test_integration_dist_tp2.py
-  - pip install -e ./plugins/vllm_add_dummy_model
-  - pytest -v -s distributed/test_distributed_oot.py
   - CUDA_VISIBLE_DEVICES=0,1 pytest -v -s test_sharded_state_loader.py
   - CUDA_VISIBLE_DEVICES=0,1 pytest -v -s kv_transfer/disagg_test.py
 
+- label: Plugin Tests (2 GPUs) # 40min
+  working_dir: "/vllm-workspace/tests"
+  num_gpus: 2
+  fast_check: true
```

Comment:
I make this new test fast check, because I assume many PRs will break it easily by importing `current_platform` too early. Adding it as fast check gives people quick signal that they should lazy import `current_platform`.

Reply:
Yeah, I think the problem now is that it's hard for users to know when the `current_platform` can be  imported globally  or lazily in future develop work. I know that all the action early than `current_platform` initiation should be lazy import. while for other developers, once they hit this kind of problem, it's a little complex and hard for them to debug. Any note or check can be added? Otherwise the PR LGTM.

Reply:
I prefer the approach in #11222 where we essentially lazily evaluate the attributes of `current_platform` even if it's imported early.

Reply:
I have to point out that my PR https://github.com/vllm-project/vllm/pull/11222 has a disadvantage that `current_platform` should not be called directly in global namespace, for example https://github.com/vllm-project/vllm/pull/11222/files#diff-3bb4c705122f0ac754ce571780b1d75d5a8d444b2c99581385c1adfaeb8562e2R302

The `__getattr__` design in this PR is a nice solution to fix the global namespace usage. https://github.com/vllm-project/vllm/pull/11602/files#diff-b353a4069aa142efe66166225c15f17b977a93fd3fc1d64482e18da127e420b8R195

So maybe another way is to combine them together, so that `current_platform` can be imported early and can be used in global namespace as well.

Reply:
I have PRs to solve the global namespace usage https://github.com/vllm-project/vllm/pull/11362. While `__getattr__ ` in this PR is a better way indeed.

Reply:
In that case, I suggest only exporting a `get_current_platform` method to indicate to other developers that `current_platform` is lazy evaluated.

Reply:
The goal of this PR (platform plugins), is to make sure, when a platform plugin is registered and should be activated, all usage of `current_platform` should resolve to that platform.

To assure this goal, I added the test https://github.com/vllm-project/vllm/pull/11602/files#diff-b147065bfd9d20787e4c9d353c862732e9d1d797d5297ca36f6b131914d2cec1 , and also make sure the `current_platform` is resolved only once.

> while for other developers, once they hit this kind of problem, it's a little complex and hard for them to debug

if they import `current_platform` too early, then this test will fail, and the error message will tell them which line to blame.

Reply:
@DarkLight1337 I think `get_current_platform` still cannot solve the `@current_platform.inference_mode()` issue because it is indeed used during module import time. To make it work with platform plugins, we have to change the code to lazy import anyway.

Reply:
Yeah, I'm just suggesting the name change to make it more obvious that `current_platform` isn't available immediately.

Reply:
>if they import current_platform too early, then this test will fail, and the error message will tell them which line to blame.
I'm OK with this. Looks clear enough

Reply:
>if they import current_platform too early, then this test will fail, and the error message will tell them which line to blame.

I'm OK with this. Looks clear enough